src/crypto/CHIPCryptoPAL.h

```diff
@@ -430,6 +432,30 @@ class HKDF_sha
                                    const uint8_t * info, size_t info_length, uint8_t * out_buffer, size_t out_length);
 };
 
+class HMAC_sha
+{
+public:
+    HMAC_sha() {}
+    virtual ~HMAC_sha() {}
+
+    /**
+     * @brief A function that implements SHA-256 based HMAC per FIPS1981.
+     *
+     * The `out_length` must be at least CHIP_CRYPTO_HASH_LEN_BYTES
+     *
+     * @param key The key to use for the HMAC operation
+     * @param key_length Length of the key
+     * @param message Message over which to compute the HMAC
+     * @param message_length Length of the message over which to compute the HMAC
+     * @param out_buffer Pointer to buffer to write output into.
+     * @param out_length Resulting length of out_buffer
```

Comment:
"Resulting" in what sense?  This is an in param...

Presumably this is just the available length of the buffer, and on success this function writes CHIP_CRYPTO_HASH_LEN_BYTES bytes to the buffer?